ntbs-service/Services/NotificationService.cs

```diff
@@ -265,13 +267,28 @@ private async Task UpdateDatabase(AuditType auditType = AuditType.Edit)
         }
 
         public IQueryable<Notification> FilterById(IQueryable<Notification> IQ, string IdFilter) {
-            return IQ.Where(s => s.NotificationId.Equals(Int32.Parse(IdFilter)) 
+            int parsedIdFilter;
+            int.TryParse(IdFilter, out parsedIdFilter);
+            return IQ.Where(s => s.NotificationId.Equals(parsedIdFilter) 
                     || s.ETSID.Equals(IdFilter) || s.LTBRID.Equals(IdFilter) || s.PatientDetails.NhsNumber.Equals(IdFilter));
         }
 
         public IQueryable<Notification> OrderQueryableByNotificationDate(IQueryable<Notification> query) {
             return query.OrderByDescending(n => n.CreationDate)
                 .OrderByDescending(n => n.SubmissionDate);
         }
+
+        public async Task<IEnumerable<Notification>> GetNotificationsWithDetailsById(IList<int> ids)
+        {
+            return await repository.GetNotificationsWithDetailsById(ids);
+        }
+
+        public async Task<List<int>> GetPaginatedIdsAsync(
```

Comment:
Also, I think you could make this generic instead of tied to `int` basically for free